data-prepper-plugins/otel-proto-common/src/test/java/org/opensearch/dataprepper/plugins/otel/codec/OTelProtoCodecTest.java

```diff
@@ -1318,4 +1318,61 @@ public void testBoundsKeyNotEqualsSign() {
         assertNotEquals(k1, k2);
     }
 
+    @Test
+    void testConvertKeysOfDataPointAttributesRemoveDuplicatedKey() {
+        final String keyName = "duplicate_key_name";
+        final String keyValue = "duplicate_value";
+        final KeyValue duplicateAttribute1 = KeyValue.newBuilder().setKey(keyName).setValue(AnyValue.newBuilder()
+                .setStringValue(keyValue).build()).build();
+        final KeyValue duplicateAttribute2 = KeyValue.newBuilder().setKey(keyName).setValue(AnyValue.newBuilder()
+                .setStringValue(keyValue).build()).build();
+        final NumberDataPoint actual = NumberDataPoint.newBuilder()
+                .addAllAttributes(Arrays.asList(duplicateAttribute1, duplicateAttribute2)).build();
+        
+        Map<String, Object> map = OTelProtoCodec.convertKeysOfDataPointAttributes(actual);
+        assertThat(map.size(), is(equalTo(1)));
```

Comment:
I understand, that it is enough to test for non-duplication. However, this is unspecified behaviour, when the key values are different. It is not clear, which value will be chosen. Even if this is undocumented behaviour, I think the unit test should force a specific selection. This is how I understand @dlvenable comment: https://github.com/opensearch-project/data-prepper/pull/4394#issuecomment-2039952138